Expose a project labeling option for "labeling engine version"

nyalldawson opened the pull request:
Adds the underlying API (currently unused) for specifying a labeling engine placement version to use for a project, and the corresponding setting in the project labeling setting dialog.

This allows users to upgrade existing projects to newer labeling engine versions. A warning will show advising users that change the version will alter the placement of labels within the project.

While currently unused, this is the framework which allows us the flexibility to tweak the labeling engine costs and obstacle avoidance strategies without altering the label placement in existing projects.

nyalldawson committed:
Add API to set a label placement engine version, which dictates which
placement rules should be followed when solving the pal labeling engine
solution.

For new projects it defaults to version 2, but when loading an older project
version 1 is used.

This allows changes to be made to the label placement problem solving without
affecting the placement of labels in existing projects.

nyalldawson committed:
[needs-docs] Add labeling engine version setting in project labeling settings dialog

This allows users to upgrade existing projects to the newer labeling engine version.
A warning will show advising users that change the version will alter the placement
of labels within the project.

nirvn commented on src/ui/qgslabelengineconfigdialog.ui:
```diff
@@ -11,7 +11,7 @@
   </property>
   <property name="windowTitle">
-   <string>Automated Placement Engine</string>
+   <string>Project Label Settings</string>
```
Isn't it the case that the dialog is also linked to diagram placement? If so, might be good to have a more generic title. The dialog _can_ be opened via the diagram's panel / dialog too.

nyalldawson replied:
Ok, reverted

nyalldawson committed:
Flip logic and leave default labeling engine as v1, mark v2 as experimental

Allows us time to tweak and refine the new rules before deploying on a
widespread basis

nyalldawson committed:
Avoid trailing ; warning thrown by QGSCOMPARENEAR macro

nyalldawson committed:
Fix method name
